### PR TITLE
Adapt @show_special to fallback to the dispatch without mime-type...

### PR DESCRIPTION
**Pull request by lgoettgens**
...if not applicable

Furthermore some doc fixes and enhancements.

cc @fingolfin

### CODE REVIEW

File: docs/src/misc.md [hunk 1]
```diff
@@ -74,16 +74,32 @@ parent. The semantics are the same as for `@show_special`.
 All are supposed to be used within the usual `show` function, where `@show_special_elem`
 is only relevant for element types of algebraic structures.
 ```julia
+@attribtes MyObj
```
Comment (joschmitt) [inline docs/src/misc.md:77]:
```suggestion
@attributes MyObj
```

**Review by fingolfin — APPROVED** [review]

**Comment by fingolfin** [comment]
Is the Hecke doctest failure genuine (and perhaps due to this PR "fixing" something?), or a fluke, or something else?!

**Comment by lgoettgens** [comment]
> Is the Hecke doctest failure genuine (and perhaps due to this PR "fixing" something?), or a fluke, or something else?!

Yes, this is "fixing" some printing in Hecke and Oscar. So let's wait with this for the larger printing changes. I'll split off the documentation fixes into a separate PR.

**Comment by lgoettgens** [comment]
The one failure is https://github.com/oscar-system/Oscar.jl/issues/3576.

**Review by fingolfin — APPROVED** [review]

**Comment by fingolfin** [comment]
OscarCI#release test failing due to 
```
Starting tests for /home/runner/.julia/packages/Oscar/59YCS/test/Aqua.jl
WARNING: Method definition number_of_generators(AbstractAlgebra.PolyRing{T} where T) in module AbstractAlgebra at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/src/Poly.jl:201 overwritten in module Oscar at /home/runner/.julia/packages/Oscar/59YCS/src/Rings/mpoly-affine-algebras.jl:241.
ERROR: Method overwriting is not permitted during Module precompilation. Use `__precompile__(false)` to opt-out of precompilation.
```
which I believe is not caused by this PR. So merging anyway.

**Pull request merged**